### DISCUSSION TITLE
Mixed Effects Logistic Regression

### DISCUSSION THREAD

**Discussion by DSGlab** [SchlossLab/mikropml#342] [Q&A] [answered]
Hi,

Is there a way to incorporate random effects into a logistic model? 

Thanks,
David

**Comment by kelly-sovacool** [comment, marked as answer]
Under the hood, mikropml uses caret for model training. Any model supported by caret should also work with mikropml. However, I don't see `glmer` or anything else related to mixed-effects logistic regression in [caret's list of models](https://topepo.github.io/caret/available-models.html). If you're motivated, you might be able to create a custom model for caret to incorporate random effects: https://topepo.github.io/caret/using-your-own-model-in-train.html

**Comment by DSGlab** [comment]
That's great to know. Thank you very much for the quick response and the link to the using your own model page.

glmer does handle mixed effects models, such as:

model <- glmer(x ~ y + z + (1  | r), data = train_data, family = binomial)

where y & z are fixed effects, and r is a random effect.

Best,
David
